onboard/nav/avoidances.cpp

```diff
@@ -0,0 +1,164 @@
+/* Original Obstacle Avoidance Algorithm
+   When an obstacle is detected, the rover will drive around whichever way
+   CV decides is the fastest. If the tennisball is detected, the heirarchy has
+   two cases in which the rover will begin to turn to tennis ball:
+        1. the difference in distance (tennis ball is closer than obs)
+        2. Turning around obs is same direction to turn to tennis ball */
+/*************************************************************************/
+Original::~Original() {}
+
+/*****************************************************/
+/* Helpers */
+/*****************************************************/
+
+// Executes the logic for turning around an obstacle.
+// If the rover is turned off, it proceeds to Off.
+// If the tennis ball is detected, the rover proceeds to it.
+// If the obstacle is no longer detcted, proceed to drive around the obstacle.
+// Else continue turning around the obstacle.
+// ASSUMPTION: We assume that the obstacle is straight ahead of us,
+//             therefore we produce an underestimate for how close the waypoint is to the
+//             obstacle. This relies on using a path width no larger than what we can
+//             confidentally see to the side.
+//
+//             There is no rock that is more than 8 meters in diameter
+NavState Original::executeTurnAroundObs( Rover * mPhoebe, const rapidjson::Document& roverConfig )
+{
+
+    double cvThresh = roverConfig[ "computerVision" ][ "obstacleThresh" ].GetDouble();
+
+    //if we see the ball and are in a search state, only go to turn to ball if turning
+    //   same direction for obstcle
+    if( mPhoebe->roverStatus().currentState() == NavState::SearchTurnAroundObs &&
+        mPhoebe->roverStatus().tennisBall().found &&
+      ( cvThresh > mPhoebe->roverStatus().tennisBall().distance - 2 or  //TODO Replace cvThresh with dist to obs
+      ( mPhoebe->roverStatus().tennisBall().bearing < 0 && mPhoebe->roverStatus().obstacle().bearing < 0 ) or
+      ( mPhoebe->roverStatus().tennisBall().bearing > 0 && mPhoebe->roverStatus().obstacle().bearing > 0 ) ) )
+    {
+        return NavState::TurnToBall;
+    }
+
+    if( !mPhoebe->roverStatus().obstacle().detected )
+    {
+        double distanceAroundObs = cvThresh / cos( fabs( degreeToRadian( mOriginalObstacleAngle ) ) );
+        mObstacleAvoidancePoint = createAvoidancePoint( mPhoebe, distanceAroundObs ) ;
+
+
+        if( mPhoebe->roverStatus().currentState() == NavState::TurnAroundObs )
+        {
+            return NavState::DriveAroundObs;
+        }
+        mJustDetectedObstacle = false;
+
+        return NavState::SearchDriveAroundObs;
+    }
+
+    double obstacleBearing = mPhoebe->roverStatus().obstacle().bearing;
+
+    if (mJustDetectedObstacle && (obstacleBearing < 0) ? (mLastObstacleAngle >= 0) : (mLastObstacleAngle < 0)) {
+        obstacleBearing *= -1;
+    }
+
+    double desiredBearing = mod( mPhoebe->roverStatus().odometry().bearing_deg
```

Comment:
try not to split the first argument onto two lines unless it really helps readability